include/irr/video/IGPUVirtualTexture.h

```diff
@@ -0,0 +1,188 @@
+#ifndef __IRR_I_GPU_VIRTUAL_TEXTURE_H_INCLUDED__
+#define __IRR_I_GPU_VIRTUAL_TEXTURE_H_INCLUDED__
+
+#include <irr/asset/ICPUVirtualTexture.h>
+#include <irr/video/IGPUImageView.h>
+#include "IVideoDriver.h"
+#include <irr/asset/IAssetManager.h>
+#include <irr/video/IGPUDescriptorSet.h>
+
+namespace irr {
+namespace video
+{
+
+class IGPUVirtualTexture final : public asset::IVirtualTexture<video::IGPUImageView>
+{
+    using base_t = asset::IVirtualTexture<video::IGPUImageView>;
+
+    IVideoDriver* m_driver;
+
+protected:
+    class IGPUVTResidentStorage final : public base_t::IVTResidentStorage
+    {
+        using base_t = base_t::IVTResidentStorage;
+        using cpu_counterpart_t = asset::ICPUVirtualTexture::ICPUVTResidentStorage;
+
+        static core::smart_refctd_ptr<IGPUImage> createGPUImageFromCPU(IVideoDriver* _driver, asset::IAssetManager* _am, asset::ICPUImage* _cpuimg)
+        {
+            auto params = _cpuimg->getCreationParameters();
+            auto img = _driver->createDeviceLocalGPUImageOnDedMem(std::move(params));
+
+            auto regions = _cpuimg->getRegions();
+            if (regions.size())
+            {
+                auto texelBuf = _driver->createFilledDeviceLocalGPUBufferOnDedMem(_cpuimg->getBuffer()->getSize(), _cpuimg->getBuffer()->getPointer());
+                _driver->copyBufferToImage(texelBuf.get(), img.get(), regions.size(), regions.begin());
+            }
+            _am->convertAssetToEmptyCacheHandle(_cpuimg, core::smart_refctd_ptr(img));
```

Comment:
won't this break the existing ICPUVirtualTExture XD ?

All this effort with copy ctors for pool allocators and you do this ;)

Reply:
ehh... yea